src/v2/Components/AddressForm.tsx

```diff
@@ -170,82 +148,130 @@ export const AddressForm: React.FC<AddressFormProps> = ({
+            onChange={handleChange}
+            onBlur={handleBlur}
+            error={getErrorMessage("addressLine2")}
           />
         </Flex>
       </TwoColumnSplit>
       <TwoColumnSplit>
         <Flex flexDirection="column">
           <Input
-            id="AddressForm_city"
+            name="address.city"
             placeholder="Add city"
             title="City"
-            value={address.city}
-            onChange={changeEventHandler("city")}
-            error={getError("city")}
+            value={values?.address?.city || ""}
+            onChange={handleChange}
+            onBlur={handleBlur}
+            error={getErrorMessage("city")}
           />
         </Flex>
 
         <Flex flexDirection="column">
           <Input
-            id="AddressForm_region"
+            name="address.region"
             placeholder="Add State, province, or region"
             title="State, province, or region"
             autoCorrect="off"
-            value={address.region}
-            onChange={changeEventHandler("region")}
-            error={getError("region")}
+            value={values?.address?.region || ""}
+            onChange={handleChange}
+            onBlur={handleBlur}
+            error={getErrorMessage("region")}
           />
         </Flex>
       </TwoColumnSplit>
-      {showPhoneNumberInput && (
+      {showPhoneNumberInput && !showPhoneNumberCountry && (
         <>
           <Flex flexDirection="column">
             <Input
-              id="AddressForm_phoneNumber"
+              name="address.phoneNumber"
               title="Phone number"
               type="tel"
-              description={phoneNumberInputDescription()}
+              description={
+                !billing ? "Required for shipping logistics" : undefined
+              }
               placeholder="Add phone"
-              pattern="[^a-z]+"
-              value={address.phoneNumber}
-              onChange={changeEventHandler("phoneNumber")}
-              error={getError("phoneNumber")}
+              autoCorrect="off"
+              value={values?.address?.phoneNumber || ""}
+              onChange={handleChange}
+              onBlur={handleBlur}
+              error={getErrorMessage("phoneNumber")}
             />
           </Flex>
           <Spacer mb={2} />
         </>
       )}
+      {showPhoneNumberInput && showPhoneNumberCountry && (
+        <Flex>
+          <Box style={{ maxWidth: "35%" }}>
+            <Select
+              title="Phone number"
+              description={
+                !billing ? "Only used for shipping purposes" : undefined
+              }
+              options={countries}
+              style={{
+                letterSpacing: "1px",
+                borderRight: "none",
+              }}
+              // TODO: Replace data-testid with queryByDisplayValue() on tests
+              // when a `selected` property is implemented
+              data-testid="phoneNumberCountry"
```

Comment:
you don't need this testid here, you can `screen.getByRole("combobox")` from the tests to query this guy